tests/installation/upgrade_select.pm

```diff
@@ -39,6 +39,8 @@ sub run {
         send_key $cmd{next};
     }
     if (match_has_tag("select-for-update")) {
+        my $arch = get_var("ARCH");
+        assert_screen('select-for-update-' . "$arch");
```

Comment:
I think you can write it like assert_screen("select-for-update-$arch");

Reply:
I have tried it, but it didn't work.

Reply:
simple run on my test environment, no problem.

my $t = 'aaa';
my $a = "test-$t";
print "got a=$a";

got a=test-aaalemon@linux-hxjt:~/perl>

Please double check your code.

Reply:
I confirmed it doesn't work. When I see your comments above, I modified it as your suggestion assert_screen("select-for-update-$arch"), the verification case didn't pass, then I changed to this way. I don't want to try again.

Reply:
As discussed, I'll investigate why your method doesn't work, and won't change pr any more.

Reply:
Here is the test result:
I created a branch with your method: https://github.com/os-autoinst/os-autoinst-distri-opensuse/compare/master...chcao:test?expand=1
And here is test result: https://openqa.suse.de/tests/6220781#step/upgrade_select/2

Reply:
- assert_screen('select-for-update-$arch');
Just as what I guess, you have make it wrong as ' ' ' while not ' " ', that why it not works.